src/coreComponents/physicsSolvers/multiphysics/MultiphasePoromechanics.cpp

```diff
@@ -37,59 +37,92 @@ namespace geos
 using namespace dataRepository;
 using namespace constitutive;
 using namespace fields;
+using namespace stabilization;
 
-MultiphasePoromechanics::MultiphasePoromechanics( const string & name,
-                                                  Group * const parent )
+namespace
+{
+
+// This is meant to be specialized to work, see below
+template< typename FLOW_SOLVER > class
+  MultiphaseCatalogNames {};
+// Class specialization for a FLOW_SOLVER set to CompositionalMultiphaseBase
+template<> class MultiphaseCatalogNames< CompositionalMultiphaseBase >
+{
+public:
+  static string name() { return "MultiphasePoromechanics"; }
+};
+// Class specialization for a FLOW_SOLVER set to CompositionalMultiphaseReservoirAndWells
+template<> class MultiphaseCatalogNames< CompositionalMultiphaseReservoirAndWells< CompositionalMultiphaseBase > >
+{
+public:
+  static string name() { return CompositionalMultiphaseReservoirAndWells< CompositionalMultiphaseBase >::catalogName() + "Poromechanics"; }
+};
+}
+
+// provide a definition for catalogName()
+template< typename FLOW_SOLVER >
+string
+MultiphasePoromechanics< FLOW_SOLVER >::
+catalogName()
+{
+  return MultiphaseCatalogNames< FLOW_SOLVER >::name();
+}
+
+template< typename FLOW_SOLVER >
+MultiphasePoromechanics< FLOW_SOLVER >::MultiphasePoromechanics( const string & name,
+                                                                 Group * const parent )
   : Base( name, parent ),
   m_isThermal( 0 )
 {
-  registerWrapper( viewKeyStruct::stabilizationTypeString(), &m_stabilizationType ).
+  this->registerWrapper( viewKeyStruct::stabilizationTypeString(), &m_stabilizationType ).
```

Comment:
Are you inserting `this->` here as a stylistic preference, or is it strictly required here?  I don't know what our code style guidelines say, but we should be consistent.

Reply:
for me it just does not compile without this because of the template introduction
i believe i took that "this" way from some other solver code, see https://github.com/search?q=repo%3AGEOS-DEV%2FGEOS%20this-%3EregisterWrapper&type=code

Reply:
ahh, right, definitely required then.